apple/internal/ios_rules.bzl

```diff
@@ -186,6 +187,97 @@ def _ios_application_impl(ctx):
         binary_target[apple_common.AppleExecutableBinary],
     ] + processor_result.providers
 
+def _ios_app_clip_impl(ctx):
+    """Experimental implementation of ios_app_clip."""
+    top_level_attrs = [
+        "app_icons",
+        "launch_images",
+        "launch_storyboard",
+        "strings",
+        "resources",
+    ]
+
+    # TODO: Replace the debug_outputs_provider with the provider returned from the linking
+    # action, when available.
+    # TODO: Extract this into a common location to be reused and refactored later when we
+    # add linking support directly into the rule.
+    binary_target = ctx.attr.deps[0]
+    binary_artifact = binary_target[apple_common.AppleExecutableBinary].binary
+
+    bundle_id = ctx.attr.bundle_id
+    embeddable_targets = ctx.attr.frameworks
+
+    processor_partials = [
+        partials.app_assets_validation_partial(
+            app_icons = ctx.files.app_icons,
+            launch_images = ctx.files.launch_images,
+        ),
+        partials.apple_bundle_info_partial(bundle_id = bundle_id),
+        partials.binary_partial(binary_artifact = binary_artifact),
+        partials.bitcode_symbols_partial(
+            binary_artifact = binary_artifact,
+            debug_outputs_provider = binary_target[apple_common.AppleDebugOutputs],
+            dependency_targets = embeddable_targets,
+            package_bitcode = True,
+        ),
+        partials.clang_rt_dylibs_partial(binary_artifact = binary_artifact),
+        partials.debug_symbols_partial(
+            debug_dependencies = embeddable_targets,
+            debug_outputs_provider = binary_target[apple_common.AppleDebugOutputs],
+            package_symbols = True,
```

Comment:
@brentleyjones Have you added support for packaging symbols?

Reply:
Not explicitly. I've tried to copy whatever is being done for `ios_application` for the most part, then making sure it's embedded into the application following the same pattern as extensions. I bet I missed some of the later though.

Reply:
`debug_symbols_partial` doesn’t have a `package_symbols` parameter, that’s why I was wondering.

Reply:
Ahh, it did back where I made the patch from 😬. Removed now.

Reply:
@thii I was confused. Yes, internally we do have support for packaging symbols.